util/random/rand.go

```diff
@@ -76,3 +81,59 @@ func (r *randstream) XORKeyStream(dst, src []byte) {
 func New() cipher.Stream {
 	return &randstream{}
 }
+
+// READER_BYTES is how many bytes we expect from each source
+const READER_BYTES = 32
+
+// MIXEDRAND_HASH is the hash used in XORKeyStream to mix all entropy sources together.
+// sha512 is chosen as its digest is 64 bytes long, matching the length of blake2xb's seed.
+var MIXEDRAND_HASH = sha512.New
+
+type mixedrandstream struct {
+	Readers []io.Reader
+}
+
+func (r *mixedrandstream) XORKeyStream(dst, src []byte) {
+
+	l := len(dst)
+	if len(src) != l {
+		panic("XORKeyStream: mismatched buffer lengths")
+	}
+
+// try to read 32 bytes from all readers and write them in a buffer
+	var b bytes.Buffer
+	var nerr int
+	for _, reader := range r.Readers {
+		buff := make([]byte, READER_BYTES)
```

Comment:
no need to allocate a new buffer each time, you can reuse the same throught the iteration